### PR TITLE
Add user update command

### LINKED ISSUE #9: As a registered owner, I can update my details
...so that I can change my personalisation.

### PR DESCRIPTION
**Pull request by zhonghan721**
Closes #9

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/user/UserUpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        requireNonNull(userUpdateDescriptor);
+        this.userUpdateDescriptor = userUpdateDescriptor;
+    }
+
+    /**
+     * Executes the register user command.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return {@code CommandResult} that indicates success.
+     * @throws CommandException if the user is already logged in or the user credentials are wrong.
```
Comment (B-enguin) [inline src/main/java/seedu/address/logic/commands/user/UserUpdateCommand.java]: Are the checks for user credentials being done in this file

File: src/main/java/seedu/address/logic/commands/user/UserUpdateCommand.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+        User updatedUser = createUpdatedUser(storedUser, userUpdateDescriptor);
+
+        model.updateUser(updatedUser);
+        return new CommandResult(MESSAGE_SUCCESS, true);
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
```
Comment (B-enguin) [inline src/main/java/seedu/address/logic/commands/user/UserUpdateCommand.java]: Would this be user instead of person?

**Review by B-enguin — COMMENTED** [review]
Some general comments

File: src/main/java/seedu/address/logic/commands/user/UserUpdateCommand.java (hunk 1 above)
Comment (zhonghan721) [reply src/main/java/seedu/address/logic/commands/user/UserUpdateCommand.java]: Good spot, will update the comments accordingly

**Review by B-enguin — APPROVED** [review]
LGTM

**Review by juliusgambe — APPROVED** [review]
LGTM

**Pull request merged**